src/HttpClientConfigurator.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+namespace FAPI\PhraseApp;
+
+use Http\Client\HttpClient;
+use Http\Client\Common\PluginClient;
+use Http\Discovery\HttpClientDiscovery;
+use Http\Discovery\UriFactoryDiscovery;
+use Http\Message\Authentication\BasicAuth;
+use Http\Message\UriFactory;
+use Http\Client\Common\Plugin;
+
+/**
+ * Configure an HTTP client.
+ *
+ * @internal This class should not be used outside of the API Client, it is not part of the BC promise.
+ */
+final class HttpClientConfigurator
+{
+    /**
+     * @var string
+     */
+    private $endpoint = 'https://api.phraseapp.com';
+
+    /**
+     * @var UriFactory
+     */
+    private $uriFactory;
+
+    /**
+     * @var HttpClient
+     */
+    private $httpClient;
+
+    /**
+     * @var Plugin[]
+     */
+    private $prependPlugins = [];
+
+    /**
+     * @var Plugin[]
+     */
+    private $appendPlugins = [];
+
+    /**
+     * @var string
+     */
+    private $token;
+
+    /**
+     * @param HttpClient|null $httpClient
+     * @param UriFactory|null $uriFactory
+     */
+    public function __construct(HttpClient $httpClient = null, UriFactory $uriFactory = null, string $token)
```

Comment:
Sure, but put it before the optional parameters. Also make sure to update the doc block